local-cli/runAndroid/runAndroid.js

```diff
@@ -124,49 +212,32 @@ function buildAndRun(args) {
     // `console.log(e.stderr)`
     return Promise.reject();
   }
-
-  try {
-    const packageName = fs.readFileSync(
-      'app/src/main/AndroidManifest.xml',
-      'utf8'
-    ).match(/package="(.+?)"/)[1];
-
-    const adbPath = getAdbPath();
-
     const devices = adb.getDevices();
```

Comment:
If we look for and validate `args.deviceId` here couldn't we simply set `devices = [args.deviceId]` and not have to fork the logic - there'd be no need for `runOnSpecificDevice` at all? Particularly because at the moment `runOnSpecificDevice` skips building `gradleArgs` - it seems better to keep things consistent and avoid repetition.